src/chrome/internal/scripts/sourcesMapper.ts

```diff
@@ -0,0 +1,58 @@
+import { LineNumber, ColumnNumber } from '../locations/subtypes';
+import { SourceMap } from '../../../sourceMaps/sourceMap';
+
+export interface ISourcesMapper {
+    readonly sources: string[];
+    getSourceLocation(scriptLocation: IPositionInScript): IPositionInSource | null;
+    getScriptLocation(scriptLocation: IPositionInSource): IPositionInScript | null;
```

Comment:
The terms SourceLocation and ScriptLocation seem a bit ambiguous. You have references to "runtimeSource" in script.ts for example.

Also the source mapper already has a convention of calling them "authored" sources and "generated" sources which make sense to me. Any reason we don't stick to those terms, here and in the rest of the code?

Reply:
SourceLocation usually refers to the data that we get from the client, which might need or not to be source mapped. ScriptLocation refers to the locations relative to the script in CDTP so no mapping is needed.

The reason I don't use authored is because the way I understand it authored refers to .ts when you have them, and to the traditional .js when you don't. This concept was meant to refer to .ts only, and empty if we don't have any.

Reply:
Actually there's a minor typo here,

`getScriptLocation(scriptLocation: IPositionInSource)`

variable name is wrong

Reply:
...and actually the methods and parameters should be named `getSourcePosition`, `getScriptPosition`, etc. to be consistent with the types and to reduce confusion with the url of the source which is also considered a location

Reply:
Changed